### DISCUSSION TITLE
When doing squash merging, how do I set the author and committer separately?

### DISCUSSION THREAD

**Discussion by jmjoy** [community/community#135214] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

Now can only set both author and committer as pr submitter.

**Comment by everforyou** [comment, marked as answer]
Hello,

When performing a squash merge on GitHub, you can only set both the author and committer to be the same as the pull request submitter. Currently, GitHub does not support setting them separately during a squash merge. 

If you need different identities for the author and committer, you'll have to handle it manually after the merge by using Git commands locally. Here's how you can do it:

1. **Perform the Squash Merge Locally**:
   - Checkout the branch you want to merge into.
   - Use the following command to squash and merge:
     ```bash
     git merge --squash <branch-name>
     ```

2. **Commit with Different Author and Committer**:
   - After squashing, you can commit with different author and committer details:
     ```bash
     git commit --author="author name <author@example.com>" -m "commit message"
     ```

3. **Push Changes**:
   - Finally, push your changes to the remote repository:
     ```bash
     git push origin <branch-name>
     ```

This way, you can have different author and committer information in your commit history.

**Reply by jmjoy** [reply]
Thanks, I hope github can support this feature in the future.

**Comment by everforyou** [comment]
I'd like get to know with you. I'm brotherrabbit91 of github.

**Reply by jmjoy** [reply]
I just saw the commits for this project which are very neat, but it should be using appveyor to handle PRs.
https://github.com/mgba-emu/mgba/commits/master/

**Reply by everforyou** [reply]
Aha, AppVeyor. Thanks for letting me know.

**Reply by jmjoy** [reply]
One PR corresponds to one commit, marked with author and committer respectively, looks neat.

**Reply by everforyou** [reply]
Good experience. ... Is this repo your work?

**Reply by jmjoy** [reply]
No.

**Reply by everforyou** [reply]
I see, What's your main stack?

**Reply by jmjoy** [reply]
Web backend

**Reply by everforyou** [reply]
Great, Can we talk outside? Which chat tool are you using?

**Comment by bcmedeiros** [comment]
> Now can only set both author and committer as pr submitter.

I would love to have the author as the author of the PR, and the committer as the person merging the PR.
